packages/flutter/lib/src/material/expansion_tile.dart

```diff
@@ -527,30 +529,39 @@ class _ExpansionTileState extends State<ExpansionTile> with SingleTickerProvider
     _iconColor = _animationController.drive(_iconColorTween.chain(_easeInTween));
     _backgroundColor = _animationController.drive(_backgroundColorTween.chain(_easeOutTween));
 
-    _isExpanded = PageStorage.maybeOf(context)?.readState(context) as bool? ?? widget.initiallyExpanded;
-    if (_isExpanded) {
+    _tileController = _chooseTileController();
+    _tileController.addListener(_onToggledExpansion);
```

Comment:
HI @fischerscode 
Are you still working on this? If yes, 

1. we will need to consider performance implications of attaching listener every ExpansionTile in a list, instead of previous behavior accessing expansion tile state.

```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final _controller = ExpansionTileController();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: SingleChildScrollView(
            child: Column(
              children: [
                for (int i = 0; i < 20; i++)
                ExpansionTile(
                  controller: _controller,
                  title: const Text('Title'),
                  trailing: Icon(
                      _controller.isExpanded ? Icons.expand_less : Icons.expand_more),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
```

2. Update the `ExpansionTile` and the [example](https://github.com/flutter/flutter/blob/master/examples/api/lib/material/expansion_tile/expansion_tile.1.dart) which specifically discusses the accessing `ExpansionTileController` via `Builder`  for this new behavior.

Reply:
@TahaTesser I've not worked on this for a while, due to time constraints and the uncertainty whether such a change is actually wanted.

I'm currently not sure how to test the performance. Should I write a test according to [this documentation](https://github.com/flutter/flutter/wiki/How-to-write-a-render-speed-test-for-Flutter) (long list and toggle expansion of multiple tiles) and use it to compare the results of the current and my implementation?

Reply:
if you run the code sample with and without PR, it attaches a new listener for every instance because of the code changes in your PR.

` _tileController.addListener(_onToggledExpansion);`


if you could consider avoiding such changes for this PR, we can move this forward. As it will be very expensive to listener for hundreds of listeners as the list grows.